coordinator/src/cli.rs

```diff
@@ -34,4 +40,40 @@ impl Opts {
 
         Ok(data_dir)
     }
+
+    /// Returns a list of addresses under which the node can be reached. Note this is used for the
+    /// node announcements.
+    pub fn p2p_announcement_addresses(&self) -> Vec<NetAddress> {
+        let mut addresses: Vec<NetAddress> = vec![];
+        if !self.p2p_address.ip().is_unspecified() {
+            addresses.push(build_net_address(
+                self.p2p_address.ip(),
+                self.p2p_address.port(),
+            ));
+        } else {
+            // Announcing the node on an unspecified ip address does not make any sense.
+            tracing::warn!("Skipping node announcement on '0.0.0.0'.");
+        }
```

Comment:
🔧 Why don't we just error and crash the application. It's probably pointless to run the app if it's misconfigured.

Reply:
It does make sense to run the app on an unspecified interface (e.g. listen on any address, local host, local ip, public ip). However, this ip does not make sense to be used when announcing our node - as this address can not be used by a client. Here we need to specify the interface. (localhost, local ip or public ip).

Since I am implying the announcement node from the p2p-address, I wouldn't fail here, as this is a valid setup.

Reply:
Okay, thanks for the explanation. I think that makes sense.

Could we maybe not print a warning then? This seems like an info or a debug log to me.

Reply:
@luckysori fyi:
https://github.com/get10101/10101/pull/240/commits/efbd04ff1e687bfe9784a2d9bfd611781a3a7cc2